### ISSUE TITLE
Implement IO-Test for the MOM Operation

### ISSUE THREAD

**Issue by lumip** [qutech/special-measure.pulsecontrol#9]
The DAC driver offers an operation to obtain the raw moment of the downsampled data.
Tests for this operation for different orders of this moment should be implemented.

**Comment by pbethke** [comment]
We should be able to write "offline" tests for this, testing only the moment calculation, provided raw io works. The reason we don't have that in down sampling and repetitive signal averaging is to test the measurement windows (masks) which raw io doesn't have

**Comment by lumip** [comment]
As far as I know, the DAC-C-library performs this operations directly on the data obtained from the acquisition card. An offline test would require the library to perform the operations on data input via software. Or am I misunderstanding you here?

**Comment by pbethke** [comment]
No, you are right. I'll have a look at the C source tomorrow and check wether it is isolated enough to test on its own

**Comment by lumip** [comment]
Maybe this could be realized as a unit test in the C(++) project. Then the library needs not to expose methods to perform operations on software input and the test would not have to bother with setting up the hardware...

**Comment by pbethke** [comment]
That is exactly what i was thinking about. as long as downsampling and the different mask types work (with the hardware), the processing could be tested individually in C++.

summoning @terrorfisch: What do you think

**Comment by lumip** [comment]
Hmm, if we take that thought one step further, downsampling and repetitive signal averaging should also be testable (?) in the C++ project, if they are not too tightly coupled to the hardware input..

**Comment by pbethke** [comment]
@terrorfisch has started writing tests for the data acquisition driver.
